File: authority-portal-api-client/src/main/java/de/sovity/authorityportal/client/AuthorityPortalClient.java
package de.sovity.authorityportal.client;

import de.sovity.authorityportal.client.gen.api.UiApi;
import lombok.Value;
import lombok.experimental.Accessors;

/**
 * API Client for the sovity Dataspace Authority Portal.
 */
@Value
@Accessors(fluent = true)
public class AuthorityPortalClient {
    UiApi uiApi;

    public static AuthorityPortalClientBuilder builder() {
        return new AuthorityPortalClientBuilder();
    }
}

File: authority-portal-api-client-ts/src/AuthorityPortalClient.ts
import {Configuration, ConfigurationParameters, UiApi} from './generated';

/**
 * API Client for our sovity Authority Portal Client
 */
export interface AuthorityPortalClient {
    uiApi: UiApi;
}

/**
 * Configure & Build new Authority Portal Client
 * @param opts opts
 */
export function buildAuthorityPortalClient(
    opts: AuthorityPortalClientOptions,
): AuthorityPortalClient {
    const config = new Configuration({
        basePath: opts.backendUrl,
        credentials: 'same-origin',
        ...opts.configOverrides,
    });

    return {
        uiApi: new UiApi(config),
    };
}

/**
 * Options for instantiating an EDC API Client
 */
export interface AuthorityPortalClientOptions {
    backendUrl: string;
    configOverrides?: Partial<ConfigurationParameters>;
}
